From ffd5dfe35895e19b3f0509954d6ecaa6fb8cc8e5 Mon Sep 17 00:00:00 2001
From: Matheus Sampaio Queiroga <srherobrine20@gmail.com>
Date: Tue, 30 Jun 2026 15:57:18 -0300
Subject: [PATCH 1/4] add load env config

Signed-off-by: Matheus Sampaio Queiroga <srherobrine20@gmail.com>
---
 cmd/main.go | 1 +
 1 file changed, 1 insertion(+)

diff --git a/cmd/main.go b/cmd/main.go
index 66b2fba461..cc68abf2a5 100644
--- a/cmd/main.go
+++ b/cmd/main.go
@@ -77,6 +77,7 @@ func prepareWorkPathAndCustomConf(cmd *cli.Command) {
 		args.CustomConf = cmd.String("config")
 	}
 	setting.InitWorkPathAndCommonConfig(os.Getenv, args)
+	setting.EnvironmentToConfig(setting.CfgProvider, os.Environ()) // Add config from env
 }
 
 type AppVersion struct {
-- 
2.55.0

